crux_kv/src/lib.rs

```diff
@@ -175,6 +188,35 @@ where
             .await
             .unwrap_exists()
     }
+
+    /// List keys that start with the provided `prefix`, starting from the provided `cursor`.
+    /// Will dispatch the event with a `KeyValueResult::ListKeys { keys: Vec<String>, cursor: u64 }`
+    /// as payload.
+    pub fn list_keys<F>(&self, prefix: String, cursor: u64, make_event: F)
+    where
+        F: Fn(Result<(Vec<String>, u64), KeyValueError>) -> Ev + Send + Sync + 'static,
+    {
+        let context = self.context.clone();
+        let this = self.clone();
+
+        self.context.spawn(async move {
+            let response = this.list_keys_async(prefix, cursor).await;
+            context.update_app(make_event(response))
+        });
+    }
+
+    /// List keys that start with the provided `prefix`, starting from the provided `cursor`,
+    /// while in an async context. This is used together with [`crux_core::compose::Compose`].
```

Comment:
Might be nice to reiterate the cursor and prefix mechanics in this doc comment too